conjure-java-jackson-serialization/src/main/java/com/palantir/conjure/java/serialization/ObjectMappers.java

```diff
@@ -102,6 +102,7 @@ public static ObjectMapper withDefaultModules(ObjectMapper mapper) {
                 .registerModule(new Jdk8Module().configureAbsentsAsNulls(true))
                 .registerModule(newSafeForNewJdksAfterburnerModule())
                 .registerModule(new JavaTimeModule())
+                .registerModule(new CollectionValidationModule())
```

Comment:
if someone objects to any performance implications of this, can they register their own CollectionModule to override this and deserialize to ImmutableLists directly?

Reply:
We don't make mappers configurable in most places, if this causes performance regressions (which I don't expect, unless the case we're validating against is hit) we should fix them.